### ISSUE TITLE
Eager loading with nested resource query

### ISSUE THREAD

**Issue by SonarSoftware** [Folkloreatelier/laravel-graphql#168]
From looking at the documentation, these seem to be mutually exclusive. I had planned to implement a recursive loader in my query class, but it doesn't seem like the parent query receives all the arguments being passed into the relations, so I can't find a good way to do that. What is the recommended approach to doing this?

For example, what I want to do is something like, load all accounts, and then load their addresses, as long as the state is 'TX'. I can handle all the logic to load and populate the appropriate resources, but I can't find any way to do this without doing a single query per parent (using the 'resolveAddressesField' function in the account type.)

**Comment by SonarSoftware** [comment]
Looks like something like this is exactly what is needed: http://webonyx.github.io/graphql-php/data-fetching/#solving-n1-problem - is this possible to implement today in this library?

**Comment by SonarSoftware** [comment]
Closing this as I was able to use the deferred function to fix this. For anyone else looking, here's the gist of how I did it.

Each type holds an array of 'QueryBuffers', which build up the list of needed items inside the 'resolve' call of the relation field. These are stored on the type itself, as a protected object.

The resolve function then returns a 'GraphQL\Deferred' function that calls 'load' on the QueryBuffer. The QueryBuffer 'load' call does a request to the database with the list of previously populated IDs, if no query has yet been performed. If the query has already been performed, it just exits.
The 'Deferred' function then calls $queryBuffer->get($id) where 'id' is the key of the result that needs to be returned.

Obviously, implementing this takes quite a bit of custom logic to map to your application, but this means that all relations now self resolve with eager loading, and can have arguments fed into them.
